src/jemalloc.c

```diff
@@ -1616,7 +1616,7 @@ imalloc_body(static_opts_t *sopts, dynamic_opts_t *dopts) {
 	 * some reason.  Let's grab it right away.
 	 */
 	tsd = tsd_fetch();
-	witness_assert_lockless(tsd_tsdn(tsd));
+	witness_assert_depth(tsd_tsdn(tsd), 0);
```

Comment:
What do you think about keeping witness_assert_lockless API (simple wrapper), for the sake of a tiny bit easier to understand / use (maybe that's just me), but at the cost of a bit of redundant API.